packages/grid/x-data-grid/src/tests/columnsGrouping.DataGrid.test.tsx

```diff
@@ -0,0 +1,366 @@
+              children: [{ field: 'col1' }, { field: 'col2' }],
+            },
+            {
+              groupId: 'col3',
+              children: [
+                {
+                  groupId: 'col3bis',
+                  children: [{ field: 'col3' }],
+                },
+              ],
+            },
+          ]}
+        />,
+      );
+
+      const row2Headers = document.querySelectorAll<HTMLElement>(
+        '[aria-rowindex="2"] [role="columnheader"]',
+      );
+
+      expect(
+        Array.from(row2Headers).map((header) => header.getAttribute('aria-colspan')),
+      ).to.deep.equal(['2', '1']);
+      expect(
+        Array.from(row2Headers).map((header) => header.getAttribute('aria-colindex')),
+      ).to.deep.equal(['1', '3']);
+    });
+  });
+
+  // TODO: remove the skip. I failed to test if an error is thrown
+  // eslint-disable-next-line mocha/no-skipped-tests
+  describe.skip('error messages', () => {
+    const TestWithError = (props: TestDataGridProps) => (
+      <ErrorBoundary logger={console}>
+        <TestDataGrid {...props} />
+      </ErrorBoundary>
+    );
+
+    it('should log an error if two groups have the same id', function test() {
+      expect(() => {
+        render(
+          <TestWithError
+            nbColumns={2}
+            columnGroupingModel={[
+              {
+                groupId: 'col12',
+                children: [{ field: 'col1' }, { groupId: 'col12', children: [{ field: 'col2' }] }],
+              },
+            ]}
+          />,
+        );
+      }).toErrorDev();
+    });
+
+    it('should log an error if a columns is referenced in two groups', function test() {
+      expect(() => {
+        render(
+          <TestWithError
+            nbColumns={3}
+            columnGroupingModel={[
+              {
+                groupId: 'col12',
+                children: [{ field: 'col1' }, { field: 'col2' }],
+              },
+              {
+                groupId: 'col23',
+                children: [{ field: 'col2' }, { field: 'col3' }],
+              },
+            ]}
+          />,
+        );
+      }).toErrorDev();
+    });
+
+    it('should log an error if a group have no id', function test() {
+      expect(() => {
+        try {
+          render(
+            <TestWithError
+              nbColumns={2}
+              columnGroupingModel={[
+                {
+                  // @ts-ignore
+                  groupId: undefined,
+                  children: [
+                    { field: 'col1' },
+                    { groupId: 'col12', children: [{ field: 'col2' }] },
+                  ],
+                },
+              ]}
+            />,
+          );
+        } catch (error) {
+          console.error(error);
+        }
+      }).toErrorDev(
+        'MUI-DataGrid: an element of the columnGroupingModel does not have either `field` or `groupId`',
+      );
+    });
+
+    it('should log an warning if a group have no children', function test() {
```

Comment:
I guess we can use an arrow function here

```suggestion
    it('should log a warning if a group has no children', () => {
```

x the other cases